integration_tests/e2e/plan-summary.cy.ts

```diff
@@ -0,0 +1,70 @@
+import CreateGoal from '../pages/create-goal'
+import PlanSummary from '../pages/plan-summary'
+
+describe('View Plan Summary', () => {
+  const planSummary = new PlanSummary()
+  const createGoalPage = new CreateGoal()
+
+  beforeEach(() => {
+    cy.createSentencePlan().then(planDetails => {
+      cy.wrap(planDetails).as('planDetails')
+      cy.openSentencePlan(planDetails.oasysAssessmentPk)
+      cy.visit('/plan-summary?source=nav')
+    })
+  })
+
+  it('Creates three new goals, and moves the middle goal up', () => {
+    createGoalPage.createCompleteGoal(0)
+    createGoalPage.createCompleteGoal(1)
+    createGoalPage.createCompleteGoal(2)
+    planSummary.clickUpOnSummaryCard(1)
+
+    planSummary
+      .getSummaryCard(0)
+      .should('contain', 'Accommodation Goal 2')
+      .and('contain', 'Move goal down')
+      .and('not.contain', 'Move goal up')
+    planSummary
+      .getSummaryCard(1)
+      .should('contain', 'Accommodation Goal 1')
+      .and('contain', 'Move goal down')
+      .and('contain', 'Move goal up')
+  })
+
+  it('Creates three new goals, and moves the middle goal down', () => {
+    createGoalPage.createCompleteGoal(0)
+    createGoalPage.createCompleteGoal(1)
+    createGoalPage.createCompleteGoal(2)
+    planSummary.clickDownOnSummaryCard(1)
+
+    planSummary
+      .getSummaryCard(1)
+      .should('contain', 'Accommodation Goal 3')
+      .and('contain', 'Move goal down')
+      .and('contain', 'Move goal up')
+    planSummary
+      .getSummaryCard(2)
+      .should('contain', 'Accommodation Goal 2')
+      .and('contain', 'Move goal up')
+      .and('not.contain', 'Move goal down')
+  })
+
+  it('Adds a long goal title (less than 128 chars)', () => {
+    createGoalPage.createGoal('accommodation')
+    createGoalPage.addGoalAutoCompletionText(
+      'This is an example of an extremely long goal title, a goal title that some may say is too long. This goal is far longer.',
+    )
+    createGoalPage.selectOtherAreasOfNeedRadio('no')
+    createGoalPage.selectStartWorkingRadio('yes')
+    createGoalPage.selectAchievementDate('In 6 months')
+    createGoalPage.clickButton('Save without steps')
+
+    planSummary.getSummaryCard(0).within(() => {
+      cy.get('.govuk-summary-card__action').first().invoke('height').should('equal', 25)
```

Comment:
Is the intent here that we have a suite of layout tests powered by Cypress to test overflows?

Reply:
The Figma design specifies that the actions should stay in line with long goal titles. I wanted to capture that as part of a test. Do you have any suggestions of a better way to test that behaviour?

Reply:
1. I think the test has a typo - the selector should be against `govuk-summary-card__actions` to check that the height of the whole list is 25, not just a single item since even on the current `main` each item always has a height of 25 even if the list wraps.
2. We should check that this is the intended behaviour for tablet sizes as well as desktop since it means the goal title wraps to make way for the actions (but make the change in a future PR if necessary)
3. I think this is a case for visual regression testing against a baseline that we set per-test per-container-size rather than in raw cypress like this - I think there was some discussion of this the other week and we said we'd open a spike in tech debt?

Reply:
1. Changed the test to check the top of the first and last action elements to make sure there is no wrap.
2. https://medium.com/@pratik_shrestha/responsive-design-testing-with-cypress-a-comprehensive-guide-fc57f33134c5 shows use of viewport for after we have checked the requirement for other screen sizes.
3. https://docs.cypress.io/plugins/#visual-testing, but can remove this test if you prefer?

Reply:
I discussed this with Tom, we previously agreed we wouldn't add visual regression testing, but I'd forgotten that - sorry. Given the small number of custom components we agreed that it wasn't worth the overhead of maintaining layout-based tests in code and so I will remove this test and we've all learned a lot about cypress in the meantime :)